resolwe/flow/serializers.py

```diff
@@ -145,24 +157,26 @@ class Meta:
         fields = ('slug', 'name', 'contributor', 'input', 'output', 'descriptor_schema',
                   'descriptor', 'tags') + update_protected_fields + read_only_fields
 
-    def __init__(self, *args, **kwargs):
-        """Initialize attributes."""
-        super(DataSerializer, self).__init__(*args, **kwargs)
-
-        request = kwargs.get('context', {}).get('request', None)
+    def get_fields(self):
+        """Dynamically adapt fields based on the current request."""
+        fields = super(DataSerializer, self).get_fields()
 
-        if not hasattr(request, 'method') or request.method == "GET":
-            self.fields['descriptor_schema'] = DescriptorSchemaSerializer(required=False)
+        if not hasattr(self.request, 'method') or self.request.method == "GET":
```

Comment:
Is `not hasattr(self.request, 'method')` still required? This was a workaround while observers didn't pass the context.

Same applies for other serializers.